### PR TITLE
[App] Breaking change update Managed Identity

### PR DESCRIPTION
**Pull request by lijinpei2008**
## Description
https://github.com/Azure/azure-powershell/pull/26083

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/App/App.Autorest/README.md [hunk 1]
```diff
@@ -62,8 +62,10 @@ identity-correction-for-post: true
 resourcegroup-append: true
 nested-object-to-string: true
 auto-switch-view: false
+# Disable transform IdentityType as GET+PUT can not replace patch
+# 1. Organizations_CreateOrUpdate can not update resources
+# 2. the input schemas of PUT and PATCH are different
 disable-transform-identity-type: true
```
Comment (vidai-msft) [inline src/App/App.Autorest/README.md:68]: Can you elaborate why this attribute is not removed?

**Comment by vidai-msft** [comment]
@lijinpei2008 Please leave a comment as change log.

File: src/App/App.Autorest/README.md [hunk 2]
```diff
@@ -675,21 +677,11 @@ directive:
 
 ##### announce upcoming MI-related breaking changes
```
[the remark this answers is not in the document]
Comment (lijinpei2008) [reply src/App/App.Autorest/README.md]: Remove this line.

**Comment by vidai-msft** [comment]
@lijinpei2008 A new autorest package has been rolled out to resolve the lower case issue. You can try to get it and re-build this module.

File: src/App/App.Autorest/README.md (hunk 1 above)
Comment (vidai-msft) [reply src/App/App.Autorest/README.md:68]: cc @dolauli

Comment (dolauli) [reply src/App/App.Autorest/README.md:68]: LGTM

File: src/App/App.Autorest/custom/New-AzContainerApp.ps1 [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+# is regenerated.
+# ----------------------------------------------------------------------------------
+
+<#
+.Synopsis
+Create a Container App.
+.Description
+Create a Container App.
+.Example
+{{ Add code here }}
```
Comment (vidai-msft) [inline src/App/App.Autorest/custom/New-AzContainerApp.ps1:23]: Please fill in examples.

File: src/App/App.Autorest/custom/New-AzContainerAppJob.ps1 [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+# is regenerated.
+# ----------------------------------------------------------------------------------
+
+<#
+.Synopsis
+Create a Container Apps Job.
+.Description
+Create a Container Apps Job.
+.Example
+{{ Add code here }}
```
Comment (vidai-msft) [inline src/App/App.Autorest/custom/New-AzContainerAppJob.ps1:23]: Please fill in examples.

File: src/App/App.Autorest/custom/Update-AzContainerApp.ps1 [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+# is regenerated.
+# ----------------------------------------------------------------------------------
+
+<#
+.Synopsis
+Patches a Container App using JSON Merge Patch
+.Description
+Patches a Container App using JSON Merge Patch
+.Example
+{{ Add code here }}
```
Comment (vidai-msft) [inline src/App/App.Autorest/custom/Update-AzContainerApp.ps1:23]: Please fill in examples.

File: src/App/App.Autorest/custom/Update-AzContainerAppJob.ps1 [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+
+<#
+.Synopsis
+Patches a Container Apps Job using JSON Merge Patch
+.Description
+Patches a Container Apps Job using JSON Merge Patch
+.Example
+{{ Add code here }}
+.Example
+{{ Add code here }}
```
Comment (vidai-msft) [inline src/App/App.Autorest/custom/Update-AzContainerAppJob.ps1:25]: Please fill in examples.

**Review by vidai-msft — CHANGES_REQUESTED** [review]

File: tools/StaticAnalysis/Exceptions/Az.App/ExampleIssues.csv [hunk 7]
```diff
@@ -74,4 +74,14 @@
...
\ No newline at end of file
+"Az.App","Update-AzContainerAppSourceControl","3","6","Mismatched_Parameter_Value_Type","5111","2","Update-AzContainerAppSourceControl -ContainerAppInputObject $containerapp is not an expected parameter value type.","-ContainerAppInputObject","Use correct parameter value type. Expected Type is Microsoft.Azure.PowerShell.Cmdlets.App.Models.IAppIdentity. Now the type is Microsoft.Azure.PowerShell.Cmdlets.App.Models.IContainerApp.(Command)."
+"Az.App","New-AzContainerAppConnectedEnvStorage","1","1","Invalid_Cmdlet","5000","1","New-AzStorageAccount is not a valid command name.","New-AzStorageAccount -ResourceGroupName azps_test_group_app -AccountName azpstestsa -Location eastus -SkuName Standard_GRS","Check the spell of New-AzStorageAccount."
```
[the remark this answers is not in the document]
Comment (lijinpei2008) [reply tools/StaticAnalysis/Exceptions/Az.App/ExampleIssues.csv:78]: When the parameter type is <IXxxxx>, to make the example clearer, I usually use the corresponding Get-AzXXXX command to get the object of the parameter, for example:

$containerapp = Get-AzContainerApp -ResourceGroupName azps_test_group_app -Name azps-containerapp-1
Update-AzContainerAppSourceControl -ContainerAppInputObject $containerapp 

This time, the variable $containerApp and the parameter ContainerAppInputObject do not have the same type in our pipeline validation rule, which will prompt the example error

File: src/App/App.Autorest/README.md [hunk 8]
```diff
@@ -62,8 +62,11 @@ identity-correction-for-post: true
...
 auto-switch-view: false
-disable-transform-identity-type: true
+# Disable transform IdentityType as GET+PUT can not replace patch(https://github.com/Azure/autorest.powershell/blob/main/docs/migration-from-v3-to-v4.md#managed-identity-best-practice-alignment)
+# 1. ContainerApps_CreateOrUpdate and Jobs_CreateOrUpdate can not update resources
+# 2. the input schemas of PUT and PATCH are different
 flatten-userassignedidentity: false
+disable-transform-identity-type: true
 
 use-extension: 
   "@autorest/powershell": "4.x"
```
[the remark this answers is not in the document]
Comment (lijinpei2008) [reply src/App/App.Autorest/README.md]: Done.

File: src/App/App.Autorest/custom/New-AzContainerApp.ps1 (hunk 3 above)
Comment (lijinpei2008) [reply src/App/App.Autorest/custom/New-AzContainerApp.ps1]: Done.

File: src/App/App.Autorest/custom/New-AzContainerAppJob.ps1 (hunk 4 above)
Comment (lijinpei2008) [reply src/App/App.Autorest/custom/New-AzContainerAppJob.ps1]: Done.

File: src/App/App.Autorest/custom/Update-AzContainerAppJob.ps1 (hunk 6 above)
Comment (lijinpei2008) [reply src/App/App.Autorest/custom/Update-AzContainerAppJob.ps1]: Done.

File: tools/StaticAnalysis/Exceptions/Az.App/ExampleIssues.csv (hunk 7 above)
[the remark this answers is not in the document]
Comment (vidai-msft) [reply tools/StaticAnalysis/Exceptions/Az.App/ExampleIssues.csv:78]: The question is why this kind of issues was detected just now but I did not see it previously. I remember no examples were changed.

[the remark this answers is not in the document]
Comment (lijinpei2008) [reply tools/StaticAnalysis/Exceptions/Az.App/ExampleIssues.csv:78]: This is a common situation, especially when providing examples of update operations using the InputObject parameter.

[the remark this answers is not in the document]
Comment (lijinpei2008) [reply tools/StaticAnalysis/Exceptions/Az.App/ExampleIssues.csv:78]: For example: 
https://github.com/Azure/azure-powershell/blob/main/tools/StaticAnalysis/Exceptions/Az.EventGrid/ExampleIssues.csv#L21

The cmdlet Update-AzEventGridClientGroup's parameter InputObject

[the remark this answers is not in the document]
Comment (lijinpei2008) [reply tools/StaticAnalysis/Exceptions/Az.App/ExampleIssues.csv:78]: I just checked and I see a lot of duplicates in there, but it's not clear to me if the validation logic has anything to do with the order.
I have an idea, I'd like to delete the ExamplesIssues.csv file and then have pipeline perform the validation and repopulate the ExamplesIssues.csv file with the returned error messages.

[the remark this answers is not in the document]
Comment (lijinpei2008) [reply tools/StaticAnalysis/Exceptions/Az.App/ExampleIssues.csv:78]: After checking, I didn't find exact duplicate lines in the ExamplesIssues.csv file, but I have the certainty that this update didn't make any changes to the example.
Why this is happening I don't understand.

**Review by vidai-msft — APPROVED** [review]
LGTM

**Pull request merged**